server/services/notifications.js

```diff
@@ -9,29 +10,21 @@ import {
   NotificationSetting,
 } from "../models";
 import { Op } from "../sequelize";
+import { createQueue } from "../utils/queue";
 
-export default class Notifications {
-  async on(event: Event) {
-    switch (event.name) {
-      case "documents.publish":
-      case "documents.update":
-        return this.documentUpdated(event);
-      case "collections.create":
-        return this.collectionCreated(event);
-      default:
-    }
-  }
-
-  async documentUpdated(event: DocumentEvent) {
-    // lets not send a notification on every autosave update
-    if (event.data && event.data.autosave) return;
+const notificationsQueue = createQueue("notifications");
 
-    // wait until the user has finished editing
-    if (event.data && !event.data.done) return;
+notificationsQueue.process(async (job) => {
+  const event = job.data;
 
+  try {
     const document = await Document.findByPk(event.documentId);
     if (!document) return;
 
+    // If the document has been updated since we initially queued a notification
+    // abort sending a notification – this functions as a debounce.
+    if (document.updatedAt > new Date(event.createdAt)) return;
```

Comment:
🍖  Compare the last updated at for the document to the event, if it was updated after the event was fired then we can discard this job.

Reply:
@thenanyu I think this is a pretty simple approach to a debounce – see anything amiss?